CustomDirective for smallrye-graphql  does not work

### Describe the bug

It seems that types of custom directive annotations are not added to the deployment, which causes an error on startup, if the directive is used in a entity.

```
import io.smallrye.graphql.api.Directive;
import io.smallrye.graphql.api.DirectiveLocation;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import org.eclipse.microprofile.graphql.NonNull;

@Directive(
    on = {DirectiveLocation.OBJECT, DirectiveLocation.INTERFACE}
)
@Retention(RetentionPolicy.RUNTIME)
public @interface CustomDirective{
    @NonNull
    String[] fields();
}
```
I'm not familar with quarkus extensions but it seems that in SmallRyeGraphQLProcessor getSchemaJavaClasses() all Types which are used in Schema are added except the Types of the Directive. I guess a single line like 

```
classes.addAll(getTypeClassNames(schema.getDirectiveTypes()));
```

will fix my problem.

### Expected behavior

Quarkus is starting even if I use custom directives within my GraphQL API

### Actual behavior

~~~
2021-10-15 19:14:36,357 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2021-10-15 19:14:36,358 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:98)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:65)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:470)
	... 15 more
Caused by: java.lang.UnsupportedOperationException
	at java.base/java.util.AbstractCollection.add(AbstractCollection.java:267)
	at io.quarkus.deployment.steps.SmallRyeGraphQLProcessor$buildExecutionService-167583418.deploy_2(SmallRyeGraphQLProcessor$buildExecutionService-167583418.zig:5725)
	at io.quarkus.deployment.steps.SmallRyeGraphQLProcessor$buildExecutionService-167583418.deploy(SmallRyeGraphQLProcessor$buildExecutionService-167583418.zig:50)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:439)
	... 15 more

~~~

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.2.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java
+++ b/extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java
@@ -75,6 +75,7 @@ import io.smallrye.graphql.cdi.producer.GraphQLProducer;
 import io.smallrye.graphql.schema.Annotations;
 import io.smallrye.graphql.schema.SchemaBuilder;
 import io.smallrye.graphql.schema.model.Argument;
+import io.smallrye.graphql.schema.model.DirectiveType;
 import io.smallrye.graphql.schema.model.Field;
 import io.smallrye.graphql.schema.model.Group;
 import io.smallrye.graphql.schema.model.InputType;
@@ -403,6 +404,7 @@ public class SmallRyeGraphQLProcessor {
         classes.addAll(getTypeClassNames(schema.getTypes().values()));
         classes.addAll(getInputClassNames(schema.getInputs().values()));
         classes.addAll(getInterfaceClassNames(schema.getInterfaces().values()));
+        classes.addAll(getDirectiveTypeClassNames(schema.getDirectiveTypes()));
 
         return classes.toArray(String[]::new);
     }
@@ -456,6 +458,16 @@ public class SmallRyeGraphQLProcessor {
         for (Type complexGraphQLType : complexGraphQLTypes) {
             classes.add(complexGraphQLType.getClassName());
             classes.addAll(getFieldClassNames(complexGraphQLType.getFields()));
+        }
+        return classes;
+    }
+
+    private Set<String> getDirectiveTypeClassNames(Collection<DirectiveType> complexGraphQLDirectiveTypes) {
+        Set<String> classes = new HashSet<>();
+        for (DirectiveType complexGraphQLDirectiveType : complexGraphQLDirectiveTypes) {
+            if (complexGraphQLDirectiveType.getClassName() != null) {
+                classes.add(complexGraphQLDirectiveType.getClassName());
+            }
         }
         return classes;
     }

--- a/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/CustomDirective.java
+++ b/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/CustomDirective.java
@@ -1,0 +1,18 @@
+package io.quarkus.smallrye.graphql.deployment;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import org.eclipse.microprofile.graphql.Description;
+import org.eclipse.microprofile.graphql.NonNull;
+
+import io.smallrye.graphql.api.Directive;
+import io.smallrye.graphql.api.DirectiveLocation;
+
+@Directive(on = { DirectiveLocation.OBJECT, DirectiveLocation.INTERFACE, DirectiveLocation.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+@Description("test-description")
+public @interface CustomDirective {
+    @NonNull
+    String[] fields();
+}

--- a/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/GraphQLTest.java
+++ b/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/GraphQLTest.java
@@ -1,7 +1,5 @@
 package io.quarkus.smallrye.graphql.deployment;
 
-import static io.quarkus.smallrye.graphql.deployment.AbstractGraphQLTest.MEDIATYPE_JSON;
-
 import java.nio.charset.Charset;
 import java.util.HashMap;
 import java.util.Map;
@@ -30,7 +28,7 @@ public class GraphQLTest extends AbstractGraphQLTest {
     static QuarkusUnitTest test = new QuarkusUnitTest()
             .withApplicationRoot((jar) -> jar
                     .addClasses(TestResource.class, TestPojo.class, TestRandom.class, TestGenericsPojo.class,
-                            BusinessException.class)
+                            CustomDirective.class, BusinessException.class)
                     .addAsResource(new StringAsset(getPropertyAsString(configuration())), "application.properties")
                     .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
 
@@ -51,6 +49,8 @@ public class GraphQLTest extends AbstractGraphQLTest {
         Assertions.assertTrue(body.contains("type TestGenericsPojo_String {"));
         Assertions.assertTrue(body.contains("enum SomeEnum {"));
         Assertions.assertTrue(body.contains("enum Number {"));
+        Assertions.assertTrue(body.contains("type TestPojo @customDirective(fields : [\"test-pojo\"])"));
+        Assertions.assertTrue(body.contains("message: String @customDirective(fields : [\"message\"])"));
     }
 
     @Test
@@ -253,6 +253,7 @@ public class GraphQLTest extends AbstractGraphQLTest {
     private static Map<String, String> configuration() {
         Map<String, String> m = new HashMap<>();
         m.put("quarkus.smallrye-graphql.events.enabled", "true");
+        m.put("quarkus.smallrye-graphql.schema-include-directives", "true");
         return m;
     }
 }

--- a/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/HotReloadTest.java
+++ b/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/HotReloadTest.java
@@ -23,7 +23,7 @@ public class HotReloadTest extends AbstractGraphQLTest {
     final static QuarkusDevModeTest TEST = new QuarkusDevModeTest()
             .withApplicationRoot((jar) -> jar
                     .addClasses(TestResource.class, TestPojo.class, TestRandom.class, TestGenericsPojo.class,
-                            BusinessException.class)
+                            CustomDirective.class, BusinessException.class)
                     .addAsResource(new StringAsset(getPropertyAsString()), "application.properties")
                     .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
 

--- a/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/TestPojo.java
+++ b/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/TestPojo.java
@@ -6,7 +6,9 @@ import java.util.List;
 /**
  * Just a test pojo
  */
+@CustomDirective(fields = "test-pojo")
 public class TestPojo {
+    @CustomDirective(fields = "message")
     private String message;
     private List<String> list = Arrays.asList("a", "b", "c");
 

